<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FindReplaceDialog</class>
 <widget class="QDialog" name="FindReplaceDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>451</width>
    <height>288</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Find and Replace</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QWidget" name="replaceBlock" native="true">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <layout class="QHBoxLayout" name="filePickerLayout">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <layout class="QFormLayout" name="formLayout">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::ExpandingFieldsGrow</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="findTextLabel">
          <property name="text">
           <string>Fi&amp;nd text:</string>
          </property>
          <property name="buddy">
           <cstring>findText</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="findText"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="replaceWithLabel">
          <property name="text">
           <string>Re&amp;place with:</string>
          </property>
          <property name="buddy">
           <cstring>replaceWithText</cstring>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="replaceWithText"/>
        </item>
        <item row="4" column="1">
         <widget class="QCheckBox" name="caseCheckBox">
          <property name="text">
           <string>Match &amp;exact case</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QCheckBox" name="wholeWordsCheckBox">
          <property name="text">
           <string>Match &amp;only whole words</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QCheckBox" name="wrapCheckBox">
          <property name="whatsThis">
           <string>When enabled, the search continues from the other end when it reaches one end of the page</string>
          </property>
          <property name="text">
           <string>&amp;Wrap around</string>
          </property>
         </widget>
        </item>
        <item row="7" column="1">
         <widget class="QCheckBox" name="backwardsCheckBox">
          <property name="whatsThis">
           <string>When set, the search goes backwards from cursor position, otherwise it goes forward</string>
          </property>
          <property name="text">
           <string>Search &amp;backwards</string>
          </property>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QCheckBox" name="regexpCheckBox">
          <property name="whatsThis">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;When checked, the pattern to find is interpreted as a UNIX regular expression. See &lt;a href=&quot;https://en.wikibooks.org/wiki/Regular_Expressions&quot;&gt;Regular Expression in Wikibooks&lt;/a&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Use regular e&amp;xpressions</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>7</number>
        </property>
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QPushButton" name="findNextButton">
          <property name="whatsThis">
           <string>Find the next occurrence from the cursor position and in the direction set by &quot;Search backwards&quot;</string>
          </property>
          <property name="text">
           <string>&amp;Find Next</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="replaceButton">
          <property name="text">
           <string>&amp;Replace</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="findAllButton">
          <property name="whatsThis">
           <string>Highlight all the occurrences of the text in the page</string>
          </property>
          <property name="text">
           <string>F&amp;ind All</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="replaceAllButton">
          <property name="whatsThis">
           <string>Replace all the occurrences of the text in the page</string>
          </property>
          <property name="text">
           <string>Replace &amp;All</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="messageLabel">
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Close|QDialogButtonBox::Help</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>findText</tabstop>
  <tabstop>replaceWithText</tabstop>
  <tabstop>caseCheckBox</tabstop>
  <tabstop>wholeWordsCheckBox</tabstop>
  <tabstop>wrapCheckBox</tabstop>
  <tabstop>backwardsCheckBox</tabstop>
  <tabstop>regexpCheckBox</tabstop>
  <tabstop>findNextButton</tabstop>
  <tabstop>replaceButton</tabstop>
  <tabstop>findAllButton</tabstop>
  <tabstop>replaceAllButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>findNextButton</sender>
   <signal>clicked()</signal>
   <receiver>FindReplaceDialog</receiver>
   <slot>findNext()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>225</x>
     <y>242</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>132</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>replaceButton</sender>
   <signal>clicked()</signal>
   <receiver>FindReplaceDialog</receiver>
   <slot>replace()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>225</x>
     <y>242</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>132</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>findAllButton</sender>
   <signal>clicked()</signal>
   <receiver>FindReplaceDialog</receiver>
   <slot>findAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>225</x>
     <y>242</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>132</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>replaceAllButton</sender>
   <signal>clicked()</signal>
   <receiver>FindReplaceDialog</receiver>
   <slot>replaceAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>225</x>
     <y>242</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>132</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>clicked(QAbstractButton*)</signal>
   <receiver>FindReplaceDialog</receiver>
   <slot>buttonBox_clicked(QAbstractButton*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>225</x>
     <y>265</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>143</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updatePreview()</slot>
  <slot>checkInput()</slot>
  <slot>updateSelection(bool)</slot>
  <slot>matchSimilar()</slot>
 </slots>
</ui>
